specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-09-01-preview/AutomationRules.json

```diff
@@ -979,11 +1091,30 @@
         }
       }
     },
+    "BooleanConditionProperties": {
```

Comment:
>BooleanConditionProperties [](http://example.com/codeflow?start=5&length=26)

These newly added properties are not being used now? What is the plan for them to be used?

Reply:
@zizw123 The AutomationRuleTriggeringLogic model holds an array called 'conditions' which is an array of objects of type AutomationRuleCondition. 
The AutomationRuleCondition model uses polymorphism and has several possible concrete types. The 2 added models in this PR (BooleanConditionProperties and PropertyArrayConditionProperties) have 'allOf' on it and are new concrete types that implement it and can be used in the conditions array. 
please notice that both BooleanConditionProperties and PropertyArrayConditionProperties have a "x-ms-discriminator-value" that corresponds to a 'conditionType' value in AutomationRuleCondition's conditionType enum

Reply:
I also updated one of the example files for this API to make the usage clearer